thumbor/handlers/__init__.py

```diff
@@ -377,7 +377,22 @@ def finish_request(self, context, result_from_storage=None):
             context.config.RESULT_STORAGE_STORES_UNSAFE or not context.request.unsafe)
 
         def inner(future):
-            results, content_type = future.result()
+            try:
+                # depending of ENGINE_THREADPOOLSIZE >0 or ==0 the
+                # future.result() is calling the corresponding
+                # operation right here, so an exception can occur here
+                # or it was already been called and the result() is simply
+                # returning the already cached result
+                future_result = future.result()
+            except Exception as e:
+                future_result = e
+
+            if isinstance(future_result, Exception):
+                logger.exception(future_result)
+                self._error(500)
+                return
```

Comment:
Could you move this block to the `except` block? Also, why not keep the original code with just an added try/except?